### ISSUE TITLE
Depencendies requirements & workflow issues

### ISSUE THREAD

**Issue by jbordonado** [Automattic/woocommerce-payments#7451]
### Describe the bug

Last time we tried to bump the requirement for WC and WP, it ended up in a production incident with reports of null issues.
We're still stuck on min WC 7.6 and min WP 6.0.

We need to publish a P2 post to initiate a discussion and look at our options.

It also impacts some GitHub workflows like E2E tests or compatibility ones, that we should fix with temp workarounds if possible.

**Comment by deepakpathania** [comment]
Investigation issue. Close this when action items are finalized based on the discussion and an issue is created for the same, add that to the current sprint. @jbordonado

**Comment by jbordonado** [comment]
P2 post for discussing the version bump halt: paJDYF-aS1-p2

Several initiatives are being conducted by the platform team to improve the robustness of the plugin in the context of unmet dependencies and update the behaviour when auto-updating with unmet dependencies. See paJDYF-b3E-p2

On our end, there are still unresolved topics:

### Version support policy

Do we change the version support policy, considering most large merchants are hesitant to update quickly to a new version and tend to stay on older versions, which makes them more vulnerable to dependency requirements issues? Supporting versions < 6 months old was a suggestion, but there has yet to be an agreement. Today's policy is L-2, and given the pace of WC releases, it means versions < 2 months old versions are supported

### Minimum WC version declared VS minimum WC version used in workflows

Do we decorrelate the min WC version we support (as seen in [woocommerce-payments.php](https://github.com/Automattic/woocommerce-payments/blob/develop/woocommerce-payments.php#L11)) and the min WC version we use for running some our workflows for compatibility? Such that we don't run into problems where, e.g., one plugin bumps the minimum requirement for WC, and we don't, and it results in the workflow breaking because that plugin throws an error due to dependency requirements being unmet.

**Comment by vladolaru** [comment]
@jbordonado I understand from this and the linked discussions that for WooPayments 6.8 (at least) we won't be bumping the WP and WC minimal requirements. Is this right?

**Comment by jbordonado** [comment]
> @jbordonado I understand from this and the linked discussions that for WooPayments 6.8 (at least) we won't be bumping the WP and WC minimal requirements. Is this right?

Correct! However, we can still bump the "Tested up to" headers if necessary, as this has nothing to do with our recent incidents, and we test against the latest, beta, and nightly builds of WP/WC in our workflows. 

Example from 6.7.0: https://github.com/Automattic/woocommerce-payments/pull/7580

**Comment by vladolaru** [comment]
Ah! Good point about bumping the "Tested up to" headers. Thanks!

**Comment by vbelolapotkov** [comment]
I think this one is blocked by #7464 #7465 and probably by #7147
